lib/azure/armrest/virtual_machine_service.rb

```diff
@@ -120,12 +120,74 @@ def stop(vmname, group = configuration.resource_group)
         vm_operate('powerOff', vmname, group)
       end
 
+      # Add new data disk to vm
+      # data_disk example
+      #  {
+      #    name:"diskname",
+      #    diskSizeGB:"1",
+      #    lun: 0,
+      #    vhd: {
+      #      uri:"https://myaccount.blob.core.windows.net/vhds/vhdname.vhd"
+      #    },
+      #    createOption: "Empty"
+      #  }
+      def attach_data_disk(vmname, data_disk, rgroup = armrest_configuration.resource_group)
+        vm = get(vmname, rgroup)
+        dds = vm.properties.storage_profile.data_disks
+        if dds.empty?
+          data = build_data_disks_request([data_disk], vm.location)
+        else
+          data = build_data_disks_request(build_data_disks(dds), vm.location)
+          data[:properties][:storageProfile][:dataDisks] << data_disk
+        end
+        update(vmname, rgroup, data)
+      end
+
+      # Remove a data disk from vm data disks list
+      # Require data_disk_name, example: "datadiskname"
+      def detach_data_disk(vmname, data_disk_name, rgroup = armrest_configuration.resource_group)
+        vm = get(vmname, rgroup)
+        dds = vm.properties.storage_profile.data_disks
+        dds.delete_if {|dd| dd.name == data_disk_name}
+        data = build_data_disks_request(build_data_disks(dds), vm.location)
+        update(vmname, rgroup, data)
+      end
+
       def model_class
         VirtualMachineModel
       end
 
       private
 
+      # Build request hash from vm data disk
+      def build_data_disks(data_disks)
```

Comment:
You need to resolve all rubocop complains, for example, this method can be rewritten 
```
data_disks.collect do |dd|
  {
    :name         => dd.name,
    :diskSizeGB   => dd.disk_size_gb,
    :lun          => dd.lun,
    :vhd          => {:uri => dd.vhd.uri}
    :createOption => dd.create_option
  }
end 
```